src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/DateFiltersTests.cs

```diff
@@ -90,22 +131,84 @@ public static IEnumerable<object[]> GetInvalidTimeZoneHandling()
             yield return new object[] { @"19701231115959+0600", "abc" };
         }
 
+        public static IEnumerable<object[]> GetInvalidDataForAddSeconds()
+        {
+            yield return new object[] { @"20badInput" };
+            yield return new object[] { @"20140130080051--0500" };
+            yield return new object[] { @"2014.051-0500" };
+            yield return new object[] { @"20140130080051123+0500" };
+            yield return new object[] { @"20201" };
+            yield return new object[] { @"2020060" };
+            yield return new object[] { @"1970-01-01T00:01:00" };
+            yield return new object[] { @"1970-01-01T00:01" };
+            yield return new object[] { @"2001-01T" };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetValidDataForAddSeconds))]
+        public void GivenSeconds_WhenAddOnValidDateTime_CorrectDateTimeStringShouldBeReturned(string originalDateTime, double seconds, string timeZoneHandling, string expectedDateTime)
+        {
+            var result = Filters.AddSeconds(originalDateTime, seconds, timeZoneHandling);
+            Assert.Equal(expectedDateTime, result);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetValidLocalDateTimeForAddSecondsWithUTCTimeZoneHandling))]
+        public void GivenSeconds_WhenAddOnValidDateTime_CorrectDateTimeShouldBeReturned(string originalDateTime, double seconds, string timeZoneHandling, string expectedDateTime)
+        {
+            var result = Filters.AddSeconds(originalDateTime, seconds, timeZoneHandling);
+            var dateTimeOffset = DateTimeOffset.Parse(result);
+            dateTimeOffset = dateTimeOffset.AddHours(TimeZoneInfo.Local.GetUtcOffset(DateTime.Now).Hours - 8);
```

Comment:
If customers run the tests in their time zone, will the tests fail? Can we get this `-8` dynamically?

Reply:
Since the test result is calculate in time zone +8:00, we minus 8 hours here to make the result applicable in different time  zone.